### ISSUE TITLE
Github-Passport Re-Login: ValidationError - email address is already in use

### ISSUE THREAD

**Issue by helgetan** [angular-fullstack/generator-angular-fullstack#1848]
Item | Version
----- | -----
generator-angular-fullstack | 3.6.1
Node | 5.7.1
npm | 3.8.5
Operating System | OS X 10.11.4

Item | Answer
----- | -----
Transpiler |  TypeScript
Markup | HTML 
CSS | SCSS 
Router |  ui-router
Build Tool | Gulp
Client Tests | Jasmine
DB | MongoDB 
Auth | Y 

Hey i'm trying to use the github passport 
got it to authenticate a user and to create a new user in mongo db but when i signout and re-sign-in with the same github account i get this error:
except the github-passport extension its generator-boilerplate-code.

error-message:

```
ValidationError: The specified email address is already in use.
   at MongooseError.ValidationError (xxx/node_modules/mongoose/lib/error/validation.js:22:11)
   at model.Document.invalidate (xxx/node_modules/mongoose/lib/document.js:1347:32)
   at xxx/node_modules/mongoose/lib/document.js:1214:17
   at validate (xxx/node_modules/mongoose/lib/schematype.js:693:7)
   at xxx/node_modules/mongoose/lib/schematype.js:721:11
   at user.model.js:122:18
From previous event:
   at model.wrappedPointCut [as save] (xxx/node_modules/mongoose/lib/document.js:1828:14)
   at passport.js:30:16
   at processImmediate [as _immediateCallback] (timers.js:383:17)
From previous event:
   at Strategy._verify (passport.js:12:10)
   at xxx/node_modules/passport-oauth2/lib/strategy.js:193:24
   at xxx/node_modules/passport-github2/lib/strategy.js:121:14
   at passBackControl (xxx/node_modules/oauth/lib/oauth2.js:125:9)
   at IncomingMessage.<anonymous> (xxx/node_modules/oauth/lib/oauth2.js:143:7)
   at emitNone (events.js:85:20)
   at IncomingMessage.emit (events.js:179:7)
   at endReadableNT (_stream_readable.js:913:12)
   at _combinedTickCallback (node.js:377:13)
   at process._tickCallback (node.js:401:11)
validationerror: The specified email address is already in use.   
  at MongooseError.ValidationError (xxx/node_modules/mongoose/lib/error/validation.js:22:11)   
  at model.Document.invalidate (xxx/node_modules/mongoose/lib/document.js:1347:32)   
  at xxx/node_modules/mongoose/lib/document.js:1214:17   
  at validate (xxx/node_modules/mongoose/lib/schematype.js:693:7)   
  at xxx/node_modules/mongoose/lib/schematype.js:721:11   
  at user.model.js:122:18
From previous event:   at model.wrappedPointCut [as save] (xxx/node_modules/mongoose/lib/document.js:1828:14)   
at passport.js:30:16   
at processImmediate [as _immediateCallback] (timers.js:383:17)From previous event:   
at Strategy._verify (passport.js:12:10)   
at xxx/node_modules/passport-oauth2/lib/strategy.js:193:24   
at xxx/node_modules/passport-github2/lib/strategy.js:121:14   
at passBackControl (xxx/node_modules/oauth/lib/oauth2.js:125:9)  
at IncomingMessage.<anonymous> (xxx/node_modules/oauth/lib/oauth2.js:143:7)   
at emitNone (events.js:85:20)   at IncomingMessage.emit (events.js:179:7)   
at endReadableNT (_stream_readable.js:913:12)   
at _combinedTickCallback (node.js:377:13)   
at process._tickCallback (node.js:401:11)
```

**Comment by Koslun** [comment]
That error is from trying to create a new user. So instead of trying to sign-in with the github account with that email, it tries to sign-up.

Not sure if there needs to be clearer separation between signing-in and signing-up buttons or if the those kinds of integrations should always first check if the email is already in use and then just sign-in rather than sign-up.

**Comment by Awk34** [comment]
Does adding this fix it?: https://github.com/angular-fullstack/generator-angular-fullstack/blob/canary/templates/app/server/api/user(auth)/user.model(mongooseModels).js#L99-L101

**Comment by helgetan** [comment]
yep this fixes it, and has already been solved with #1851
